### PR TITLE
feat(clickhouse sink): Add support for basic auth

### PR DESCRIPTION
**Pull request by amuluowin**
Update the code and docs

### CODE REVIEW

**Comment by amuluowin** [comment]
This is my first time using ruby, after `make generate` a lot of docs changed, i'm not sure if it's right.

**Comment by amuluowin** [comment]
This is my first time using ruby, after `make generate` a lot of docs has changed, i'm not sure if it's right.

File: src/sinks/clickhouse.rs [hunk 1]
```diff
@@ -36,13 +37,15 @@ pub struct ClickhouseConfig {
     pub request_rate_limit_num: Option<u64>,
     pub request_retry_attempts: Option<usize>,
     pub request_retry_backoff_secs: Option<u64>,
+
+    pub basic_auth: Option<ClickHouseBasicAuthConfig>,
```
Comment (LucioFranco) [inline src/sinks/clickhouse.rs]: minor nit: can we move this struct item to live under `compression`?

**Comment by LucioFranco** [comment]
Other this LGTM besides the docs issue.

File: src/sinks/clickhouse.rs (hunk 1 above)
Comment (binarylogic) [reply src/sinks/clickhouse.rs]: @LucioFranco can you resolve this so we can merge?

Comment (LucioFranco) [reply src/sinks/clickhouse.rs]: Not blocking we can merge anyways

**Pull request merged**